generic-server/src/plugin/container.rs

```diff
@@ -89,6 +88,35 @@ impl<'a> PluginContainer<'a> {
         }
     }
 
+    pub fn unload(&mut self) -> Result<(), PluginContainerError> {
+        // Unmount plugins and clearing the vector of routes
+        let errors: HashMap<_, _> = self
+            .plugins
+            .iter()
+            .filter_map(|plugins| match plugins.unmount() {
+                Ok(_) => {
+                    tracing::info!("Unmounted plugins {}", plugins.name());
+                    self.collected_routes.clear();
+                    None
+                }
+                Err(err) => {
+                    tracing::error!("error unmounting plugins {}", plugins.name());
+                    Some((plugins.name().to_owned(), err))
+                }
+            })
+            .collect();
+
+        // Flag as unloaded
+        self.loaded = false;
+
+        // Return state of completion
+        if errors.is_empty() {
+            tracing::debug!("plugin container unloaded");
+            Ok(())
+        } else {
+            Err(PluginContainerError::PluginErrorMap(errors))
+        }
+    }
     /// Merge collected routes from all plugins successfully initialized.
     pub fn routes(&self) -> Result<Router, PluginContainerError> {
```

Comment:
Add a space after the function. Also, think about auto-formatting the file. In vscode: Right Click + Format Document.